app/uk/gov/hmrc/eoricommoncomponent/frontend/services/CheckYourEmailService.scala

```diff
@@ -0,0 +1,175 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.eoricommoncomponent.frontend.services
+
+import play.api.Logging
+import play.api.data.Form
+import play.api.mvc.{AnyContent, MessagesControllerComponents, Request, Result}
+import uk.gov.hmrc.eoricommoncomponent.frontend.controllers.CdsController
+import uk.gov.hmrc.eoricommoncomponent.frontend.controllers.email.routes.{
+  CheckYourEmailController,
+  WhatIsYourEmailController
+}
+import uk.gov.hmrc.eoricommoncomponent.frontend.controllers.routes.{
+  EmailController,
+  SecuritySignOutController,
+  UserLocationController
+}
+import uk.gov.hmrc.eoricommoncomponent.frontend.domain.{GroupId, LoggedInUserWithEnrolments, YesNo}
+import uk.gov.hmrc.eoricommoncomponent.frontend.forms.models.email.EmailForm.confirmEmailYesNoAnswerForm
+import uk.gov.hmrc.eoricommoncomponent.frontend.models.Service
+import uk.gov.hmrc.eoricommoncomponent.frontend.services.cache.SessionCache
+import uk.gov.hmrc.eoricommoncomponent.frontend.services.email.EmailVerificationService
+import uk.gov.hmrc.eoricommoncomponent.frontend.views.html.email.{check_your_email, email_confirmed, verify_your_email}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class CheckYourEmailService @Inject() (
+  save4LaterService: Save4LaterService,
+  emailVerificationService: EmailVerificationService,
+  sessionCache: SessionCache,
+  mcc: MessagesControllerComponents,
+  checkYourEmailView: check_your_email,
+  verifyYourEmail: verify_your_email,
+  emailConfirmedView: email_confirmed
+)(implicit ec: ExecutionContext)
+    extends CdsController(mcc) with Logging {
+
+  def fetchEmailAndPopulateView(
+    userWithEnrolments: LoggedInUserWithEnrolments,
+    service: Service,
+    emailVerificationView: Boolean = false
+  )(implicit request: Request[AnyContent]): Future[Result] =
+    save4LaterService.fetchEmail(GroupId(userWithEnrolments.groupId)) flatMap {
+      _.fold {
+        if (emailVerificationView) {
+          // $COVERAGE-OFF$Loggers
+          logger.warn(s"[CheckYourEmailService][verifyEmailView] -   emailStatus cache none")
+          // $COVERAGE-ON
+          Future.successful(Ok(verifyYourEmail(None, service)))
+        } else {
+          // $COVERAGE-OFF$Loggers
+          logger.warn(s"[CheckYourEmailService][createForm] -   emailStatus cache none")
+          // $COVERAGE-ON
+          Future.successful(Ok(checkYourEmailView(None, confirmEmailYesNoAnswerForm, isInReviewMode = false, service)))
+        }
+      } { emailStatus =>
+        if (emailVerificationView)
+          Future.successful(Ok(verifyYourEmail(emailStatus.email, service)))
+        else
+          Future.successful(
+            Ok(checkYourEmailView(emailStatus.email, confirmEmailYesNoAnswerForm, isInReviewMode = false, service))
+          )
+      }
+    }
```

Comment:
It almost feels like the above could be two methods, fetchEmailAndPopulateVerificationView and fetchEmailAndPopulateCheckView which may be tidier and easier to read